nixos/modules/services/web-apps/rundeck.nix

```diff
@@ -0,0 +1,550 @@
+        };
+
+        host = lib.mkOption {
+          type = lib.types.str;
+          default = "localhost";
+          description = "Database host";
+        };
+
+        port = lib.mkOption {
+          type = lib.types.port;
+          default = 5432;
+          description = "Database port";
+        };
+
+        name = lib.mkOption {
+          type = lib.types.str;
+          default = "rundeck";
+          description = "Database name";
+        };
+
+        username = lib.mkOption {
+          type = lib.types.str;
+          default = "rundeck";
+          description = "Database username";
+        };
+
+        password = lib.mkOption {
+          type = lib.types.str;
+          default = "rundeck";
+          description = "Database password";
+        };
+      };
+
+      settings = lib.mkOption {
+        type = lib.types.attrsOf (
+          lib.types.oneOf [
+            lib.types.bool
+            lib.types.int
+            lib.types.str
+          ]
+        );
+        default = { };
+        description = ''
+          Rundeck configuration options.
+          See https://docs.rundeck.com/docs/administration/configuration/config-file-reference.html
+          for all available options.
+        '';
+        example = lib.literalExpression ''
+          {
+            "server.address" = "0.0.0.0";
+            "server.port" = "4440";
+            "grails.serverURL" = "http://rundeck.example.com:4440";
+            "rundeck.projectsStorageType" = "db";
+          }
+        '';
+      };
+
+      ssl = {
+        enable = lib.mkEnableOption "SSL support";
+
+        keyStore = lib.mkOption {
+          type = lib.types.path;
+          example = "/etc/rundeck/ssl/keystore";
+          description = "Path to the keystore containing the SSL certificate";
+        };
+
+        keyStorePassword = lib.mkOption {
+          type = lib.types.str;
+          description = "Password for the SSL keystore";
+        };
+
+        keyPassword = lib.mkOption {
+          type = lib.types.str;
+          description = "Password for the SSL key";
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.database.type == "h2" -> true;
+        message = "When using H2 database, no additional configuration is needed";
+      }
+      {
+        assertion = cfg.database.type != "h2" -> cfg.database.password != "";
+        message = "Database password must be set when not using H2 database";
+      }
+    ];
+
+    users.users.${cfg.user} = {
+      isSystemUser = true;
+      group = cfg.group;
+      home = cfg.dataDir;
+      createHome = true;
+    };
+
+    users.groups.${cfg.group} = { };
+
+    systemd.tmpfiles.settings."10-rundeck" = {
+      "${cfg.dataDir}" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/etc" = {
```

Comment:
Is this not /etc/... ?

Reply:
Rundeck uses its tree structure.
And I want to be close to what is generated by rundeck